src/Common/tests/Cryptography/AlgorithmImplementations/RSA/RSAFactory.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Security.Cryptography.Rsa.Tests
+{
+    public interface IRSAProvider
+    {
+        RSA Create();
+        RSA Create(int keySize);
```

Comment:
I'm fine with this approach for the case where we're really testing an individual library with shared code. 

The 4.6 tests in question also check interoperability between multiple implementations and so I also think we'll need to solve the "where to put it" and have a downstream test library eventually... Let's not block this change on that, though.